crates/sui-core/src/consensus_validator.rs

```diff
@@ -71,7 +72,11 @@ impl TransactionValidator for SuiTxValidator {
                 ConsensusTransactionKind::UserTransaction(certificate) => {
                     self.metrics.certificate_signatures_verified.inc();
                     certificate.data().verify()?;
-                    let idx = obligation.add_message(certificate.data(), certificate.epoch());
+                    let idx = obligation.add_message(
+                        certificate.data(),
+                        certificate.epoch(),
+                        Intent::default().with_scope(IntentScope::SenderSignedTransaction),
```

Comment:
here we can simplify without passing in intent, because add_message<T> where T implements Serialize only, but not Message. We can alternatively make T to implement Serialize + Message, but that involves a bigger refactor and make the interface a bit heavy since now everything implements Message